tests/gcm/test_distribution_change_robust.py

```diff
@@ -0,0 +1,56 @@
+import networkx as nx
+import numpy as np
+import pandas as pd
+from flaky import flaky
+from pytest import approx
+
+from dowhy import gcm
+
+
+def _gen_data(N=10000):
+    X1_old = np.random.normal(1, 1, N)
+    X2_old = 0.5 * X1_old + np.random.normal(0, 1, N)
+    Y_old = X1_old + X2_old + 0.25 * X1_old**2 + 0.25 * X2_old**2 + np.random.normal(0, 1, N)
+    X1_new = np.random.normal(1, 1.1, N)
+    X2_new = 0.2 * X1_new + np.random.normal(0, 1, N)
+    Y_new = X1_new + X2_new + 0.25 * X1_new**2 - 0.25 * X2_new**2 + np.random.normal(0, 1, N)
+    data_old = pd.DataFrame({"X1": X1_old, "X2": X2_old, "Y": Y_old})
+    data_new = pd.DataFrame({"X1": X1_new, "X2": X2_new, "Y": Y_new})
+    return data_old, data_new
+
+
+@flaky(max_runs=5)
+def test_given_two_data_sets_with_different_mechanisms_when_evaluate_distribution_change_then_returns_expected_result_mean():
+    data_old, data_new = _gen_data()
+    causal_model = gcm.ProbabilisticCausalModel(nx.DiGraph([("X1", "X2"), ("X1", "Y"), ("X2", "Y")]))
+    shap = gcm.distribution_change_robust(
+        causal_model,
+        data_old,
+        data_new,
+        "Y",
+        regressor=gcm.ml.regression.create_knn_regressor,
+        xfit_folds=10,
+    )
+
+    assert shap["X1"] == approx(0.054, abs=0.1)
+    assert shap["X2"] == approx(-0.298, abs=0.1)
+    assert shap["Y"] == approx(-0.651, abs=0.1)
```

Comment:
I am wondering, in theory, the old `distribution_change` method should also give the same results (just higher variance), right? Wondering if it makes sense to add one more test to just see that both methods approximately return the same values.

Reply:
In theory, they are both consistent for the same object, but the MR method has a weakly faster rate of convergence. In practice, `distribution_change` seems to work very well for the mean, but it seems to be a bit far off for the variance in this test.

Reply:
How much is "far off" and is it also far off with more data etc.? Just want to avoid that there is a bug in `distribution_change` :)

Reply:
MAEs between .15 and .6 for $N = 10,000$. It does get better for larger $N$ (with $N = 100,000$ I get MAEs between .11 and .4).

Reply:
N=10000 seems already quite large for this rather simple SCM. That being said, I assume it is rather due to the hyperparameters (e.g., too few samples from the modified distribution etc.). I can take a look at it separately. Just in case, these expected variances changes (0.054, -0.298 etc.) are the "ground truth" ones?

Reply:
The true shap values for the variance change are 0.811, -1.343, -1.397 for X1, X2 and Y respectively.

Reply:
If this is the case, why are the results that far off? Wouldn't this indicate that something if wrong?

Reply:
If I understand your question correctly, the numbers above

```
    assert shap["X1"] == approx(0.054, abs=0.1)
    assert shap["X2"] == approx(-0.298, abs=0.1)
    assert shap["Y"] == approx(-0.651, abs=0.1)
```

are for the change in mean. `distribution_change` seems to be working well for the change in mean.

The Shapley values for the change in variance are the next function in the same file `test_distribution_change_robust.py`, the ground truth is 0.811, -1.343, -1.397.

Reply:
Ah yes sorry, didn't look at the context of the code again. Never mind then!